core/trino-main/src/main/java/io/trino/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -872,14 +869,10 @@ public HashComputationSet withHashComputation(Optional<HashComputation> hashComp
             return Optional.empty();
         }
 
-        Expression result = new GenericLiteral(StandardTypes.BIGINT, String.valueOf(INITIAL_HASH_VALUE));
```

Comment:
Does it have to be in sync with `io.trino.operator.HashAggregationOperator#calculateDefaultOutputHash`?

Reply:
In terms of functionality no. My change does not change anything other than maybe perf.

Given how `io.trino.operator.scalar.CombineHashFunction#getHash` is implemented now, we can greatly simplify `HashAggregationOperator#calculateDefaultOutputHash` though. 
Unless I'm mistaken, this currently returns `BigintType.hash(groupId) * Math.pow(31, groupByChannels.size()-groupIdChannel -1)`.